core/src/main/java/bisq/core/xmr/knaccc/monero/address/WalletAddress.java

```diff
@@ -121,7 +123,21 @@ public static String getSubaddressBase58(Scalar privateViewKey,
 
     }
 
-    public String getSubaddressBase58(String privateViewKeyHex, long accountId, long subaddressId) {
+    public String getSubaddressBase58(String privateViewKeyHex, long accountId, long subaddressId) throws InvalidWalletAddressException {
+        if (!checkPrivateViewKey(privateViewKeyHex)) {
+            throw new InvalidWalletAddressException("Wrong private view key for main address");
+        }
         return getSubaddressBase58(new Scalar(privateViewKeyHex), hexToBytes(getPublicSpendKeyHex()), accountId, subaddressId);
     }
+
+    public boolean checkPrivateViewKey(String privateViewKey) {
+        return arePubPrivKeysRelated(this.publicViewKeyHex, privateViewKey);
+    }
+
+    public static boolean arePubPrivKeysRelated(String publicKey, String privateKey) {
```

Comment:
As far I understand the pubKey must match the derived pubKey from the privKey. If that is correct I would prefer a more precise method name. isRelated is more broad as the code suggests.